src/main/java/com/yelp/nrtsearch/server/field/PrefixFieldDef.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2025 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.field;
+
+import com.yelp.nrtsearch.server.analysis.PrefixWrappedAnalyzer;
+import com.yelp.nrtsearch.server.grpc.Field;
+import com.yelp.nrtsearch.server.grpc.PrefixQuery;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.document.FieldType;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.*;
+import org.apache.lucene.util.automaton.Automata;
+import org.apache.lucene.util.automaton.Automaton;
+import org.apache.lucene.util.automaton.Operations;
+
+public class PrefixFieldDef extends TextBaseFieldDef {
+  private final int minChars;
+  private final int maxChars;
+  private final String parentField;
+  private static final String INDEX_PREFIX = "._index_prefix";
+
+  public PrefixFieldDef(
+      String parentName, Field requestField, FieldDefCreator.FieldDefCreatorContext context) {
+    super(parentName + INDEX_PREFIX, requestField, context);
+    this.minChars = requestField.getIndexPrefixes().getMinChars();
+    this.maxChars = requestField.getIndexPrefixes().getMaxChars();
+    this.parentField = parentName;
+  }
+
+  @Override
+  protected void setSearchProperties(FieldType fieldType, Field requestField) {
+    fieldType.setOmitNorms(true);
+    fieldType.setTokenized(true);
+    if (requestField.getSearch()) {
+      setIndexOptions(requestField.getIndexOptions(), fieldType, IndexOptions.DOCS);
+    }
```

Comment:
You know this field will have `search=true` and no `IndexOptions` set, so you can probably just set the value to `DOCS`